NETOBSERV-121 tab pages

jpinsonneau opened the pull request:
Add details pages for Pod / Service / Namespace / Deployment

Automatically set filters from selected object infos

![image](https://user-images.githubusercontent.com/91894519/149144485-26f4fa99-2c8e-47ec-b850-f22287dff590.png)

jotak commented:
Really nice!
I'd put it on hold in order to gather some feedback on the feature, but it's already super interesting as it is.
/hold

stleerh commented:
/lgtm

jotak commented:
/lgtm
we must create a follow-up task to finish the work once https://issues.redhat.com/browse/NETOBSERV-73 is done

jotak commented:
Follow-up created: https://issues.redhat.com/browse/NETOBSERV-156

jpinsonneau commented:
Updated from feedbacks : 
- filters cannot be updated in tabs, `chips` show current forced filters
- added 'Edit filter' button to redirect to `/netflow-traffic` page keeping url arguments (filters / refresh)
- in case of 'No result found', clear all filter button will also redirect to `/netflow-traffic` removing all arguments

jpinsonneau commented:
/retest

jotak commented on web/src/utils/router.ts:
```diff
@@ -73,3 +75,11 @@ export const removeQueryArguments = (keys: AnyQueryArgs[]) => {
   }
 };
+
+export const navigate = (pathName: string, keepParams = true) => {
```
it seems that this `navigate` triggers a full reload of the console, if possible we should probably avoid that (but i don't know if there are other options)

jotak replied:
(I think this is what react-router is for)

jpinsonneau replied:
Thanks, I used 'react-router-dom' `useHistory` hook :+1:

jotak commented on web/src/utils/router.ts:
```diff
@@ -43,13 +44,28 @@ const reporterToFlowdir = _.invert(flowdirToReporter);
   const params: QueryArguments = {};
-  _.each(filters, (f: Filter) => {
+  let allFilters: Filter[];
+  if (forcedFilters) {
```
is there cases where we have `forcedFilters` AND some extra `filters` ? I thought if not, we could simplify and, from the caller, pass either `filters` or `forcedFilters`, but not both...

jpinsonneau replied:
`FlowDirection` and `Limit` are editable filters from the tabs now; so yes you can have both !

However you are not supposed to have forced & normal filters on the same columns but the user may try to edit URL arguments right ?

jotak replied:
But `FlowDirection` and `Limit` are not passed through `filters`, but through `opts`. They should not affect the search filters

jpinsonneau replied:
You're right, I will simplify that then :+1: 

Thanks for the feedback

jotak commented:
looks like filtering by namespace is broken :/ (maybe it's me...)

jpinsonneau commented:
> looks like filtering by namespace is broken :/ (maybe it's me...)

Hmm good catch ! I also reproduce from main branch.

This seems to be related to `type QueryArguments = { [k in AnyQueryArgs]?: unknown };` that produce wrong params key `SrcNamespace[]` instead of `SrcNamespace`

jotak commented:
/lgtm
(rebased locally & tested)

jotak commented:
/unhold
